### PR TITLE
Integrate brand-consolidation with master

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
This branch was created off of `brand-consolidation`, then merged `master` via `git merge origin/master`, and is now being merged back into `master`.

The `assets` and the `va-gov` directories have already been merged into `master` so as to reduce the diff and help identify risks. The process we have followed with the remaining files is inspect each change, determine its compatibility, and either revert back to the Vets.gov `master` version or incorporate the feature flag. If there was no impact on Vets.gov/master, then no change was required.

Moving forward, we should adapt our build to use only `master`, then do away with `brand-consolidation`.

### Related pull requests
- [Merge brand-consolidation va-gov directory into master](https://github.com/department-of-veterans-affairs/vets-website/pull/8513)
- [Merge brand-consolidation assets directory into master](https://github.com/department-of-veterans-affairs/vets-website/pull/8512)

### Tickets
Resolves https://github.com/department-of-veterans-affairs/vets.gov-team/issues/12840
Resolves department-of-veterans-affairs/vets.gov-team/issues/13332.

## Testing done
- [x] Automated tests approve
- [x] Manual testing

## Screenshots
Todo

## Acceptance criteria
- [x] `master` and `brand-consolidation` have been merged into one another and are in sync. 

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/static-pages/static-pages-entry.js [hunk 1]
```diff
@@ -29,29 +54,39 @@ const store = createCommonStore();
...
   createEducationApplicationStatus(store);
 }
+
 if (location.pathname === eduOptOutPage) {
   createOptOutApplicationStatus(store);
 }
+
 if (burialPages.has(location.pathname)) {
   createApplicationStatus(store, {
     formId: '21P-530',
```
Comment (ncksllvn) [inline src/applications/static-pages/static-pages-entry.js]: Compatible ✅

File: src/applications/static-pages/sidebar-navigation.js [hunk 2]
```diff
@@ -31,15 +31,17 @@ class SideBarMenu {
...
     this.closeMenu(trigger);
   }
 
   closeMenu() {
     const close = this.menu.querySelector('.va-sidebarnav-close');
     close.addEventListener('click', () => {
       this.menu.classList.remove('va-sidebarnav--opened');
-      document.body.classList.remove('va-pos-fixed');
+      document.getElementsByTagName('body')[0].style.overflow = 'initial';
+      document.getElementsByClassName('va-btn-sidebarnav-trigger')[0].removeAttribute('hidden');
```
Comment (ncksllvn) [inline src/applications/static-pages/sidebar-navigation.js]: Made a ticket to double check this change, as it does affect Vets.gov - https://github.com/department-of-veterans-affairs/vets.gov-team/issues/13517

File: src/applications/personalization/profile360/vet360/util/local-vet360.js [hunk 3]
```diff
@@ -4,7 +4,8 @@ import * as VET360_CONSTANTS from '../constants';
 export function isVet360Configured() {
   return [
     'staging.vets.gov',
-    'www.vets.gov'
+    'www.vets.gov',
+    'preview.va.gov'
```
Comment (ncksllvn) [inline src/applications/personalization/profile360/vet360/util/local-vet360.js]: Compatible ✅

File: src/platform/site-wide/index.js [hunk 4]
```diff
@@ -40,4 +47,10 @@ export default function startSitewideComponents(commonStore) {
   startUserNavWidget(commonStore);
   startFeedbackWidget(commonStore);
   startAnnouncementWidget(commonStore);
+
+  if (brandConsolidation.isEnabled()) {
+    startMegaMenuWidget(commonStore);
+    startMobileMenuButton(commonStore);
+    startVAFooter(commonStore);
+  }
```
Comment (ncksllvn) [inline src/platform/site-wide/index.js]: Compatible ✅

File: src/platform/site-wide/user-nav/components/SearchMenu.jsx [hunk 5]
```diff
@@ -24,13 +30,13 @@ class SearchMenu extends React.Component {
...
         acceptCharset="UTF-8"
-        action="https://search.vets.gov/search"
+        action={this.state.searchAction}
         id="search"
         method="get">
         <div className="csp-inline-patch-header">
           <input name="utf8" type="hidden" value="&#x2713;"/>
         </div>
-        <input id="affiliate" name="affiliate" type="hidden" value="vets.gov_search"/>
+        <input id="affiliate" name="affiliate" type="hidden" value={this.state.searchAffiliate}/>
```
Comment (ncksllvn) [inline src/platform/site-wide/user-nav/components/SearchMenu.jsx]: Compatible ✅

File: src/platform/site-wide/user-nav/containers/Main.jsx [hunk 6]
```diff
@@ -85,6 +86,12 @@ export class Main extends React.Component {
     }
   }
 
+  bindModalTriggers = () => {
+    const triggers = document.querySelectorAll('.signin-signup-modal-trigger');
+    const openLoginModal = () => this.props.toggleLoginModal(true);
+    triggers.forEach(t => t.addEventListener('click', openLoginModal));
+  }
+
```
Comment (ncksllvn) [inline src/platform/site-wide/user-nav/containers/Main.jsx]: Compatible ✅

File: src/platform/startup/store.js [hunk 7]
```diff
@@ -23,7 +34,8 @@ export const commonReducer = {
   feedback,
   scheduledDowntime,
   announcements,
-  buildSettings
+  buildSettings,
+  ...brandConsolidatedReducers
```
Comment (ncksllvn) [inline src/platform/startup/store.js]: Compatible ✅

File: src/platform/user/authentication/utilities.js [hunk 8]
```diff
@@ -5,8 +5,19 @@ import recordEvent from '../../monitoring/record-event';
...
+  if (brandConsolidation.isPreview()) {
+    successRelay = 'preview_vagov';
+  } else {
+    successRelay = 'vagov';
+  }
+}
+
 const SESSIONS_URI = `${environment.API_URL}/sessions`;
-const redirectUrl = (type) => `${SESSIONS_URI}/${type}/new`;
+const redirectUrl = (type) => `${SESSIONS_URI}/${type}/new?success_relay=${successRelay}`;
```
Comment (ncksllvn) [inline src/platform/user/authentication/utilities.js]: Compatible, as this defaults to `vetsgov` and has already been implemented in the API ✅ 

cc @omgitsbillryan @saneshark @U-DON

File: src/platform/utilities/accessibility/index.js [hunk 9]
```diff
@@ -21,4 +22,7 @@ function getTabbableElements(node) {
   return node.querySelectorAll('input,select,a[href],textarea,button,[tabindex]');
 }
 
-export { isWideScreen, isEscape, isTab, isReverseTab, getTabbableElements };
+function isEnter(e) {
+  return (e.key === 'Enter' || e.keyCode === ENTER);
+}
+export { isWideScreen, isEscape, isTab, isReverseTab, getTabbableElements, isEnter };
```
Comment (ncksllvn) [inline src/platform/utilities/accessibility/index.js]: Compatible ✅

File: src/platform/site-wide/usa-banner-toggle.js [hunk 10]
```diff
@@ -3,21 +3,19 @@
...
   const usaHeader = document.querySelector('.usa-banner-header');
   const govBanner = document.querySelector('#gov-banner');
 
+
   toggleButton.addEventListener('mouseup', () => {
     const shouldExpand = govBanner.getAttribute('aria-hidden') === 'true';
 
     usaHeader.classList.toggle('usa-banner-header-expanded', shouldExpand);
-    toggleButton.setAttribute('aria-expanded', shouldExpand);
     govBanner.setAttribute('aria-hidden', !shouldExpand);
```
Comment (ncksllvn) [inline src/platform/site-wide/usa-banner-toggle.js]: Compatible ✅ 

cc @RobAWilkinson

File: config/webpack.config.js [hunk 11]
```diff
@@ -175,13 +175,13 @@ const configGenerator = (options, apps) => {
...
       new ExtractTextPlugin({
-        filename: (['development', 'devpreview'].includes(options.buildtype)) ? '[name].css' : `[name].[contenthash]-${timestamp}.css`
+        filename: (['development', 'vagovdev'].includes(options.buildtype)) ? '[name].css' : `[name].[contenthash]-${timestamp}.css`
       }),
       new webpack.IgnorePlugin(/^\.\/locale$/, /moment$/),
     ],
   };
 
-  if (['production', 'staging', 'preview'].includes(options.buildtype)) {
+  if (['production', 'staging', 'preview', 'vagovstaging'].includes(options.buildtype)) {
```
Comment (ncksllvn) [inline config/webpack.config.js]: Compatible ✅ 

cc @wyattwalter

**Comment by ncksllvn** [comment]
All tests are currently passing. Remaining items include: 
- [ ] Ensure the compatibility of our SASS changes with Vets.gov
- [ ] [Adapt the `Jenkinsfile` to build Preview off of `master`](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/13512)
    - cc @wyattwalter
- [ ] Manual testing for Vets.gov
- [ ] Manual testing for Preview
    - E2E tests do not execute for Preview, so we mainly need to double check that all application `rootUrl`s are being derived properly.

cc @jkassemi

**Review by jbalboni — APPROVED** [review]
LGTM (reviewed primarily in calls between Nick and I)

**Pull request merged**